src/cascadia/TerminalSettingsEditor/GlobalAppearance.cpp

```diff
@@ -28,4 +45,130 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     }
+
+    winrt::hstring GlobalAppearance::LanguageDisplayConverter(const winrt::hstring& tag)
+    {
+        if (tag == systemLanguageTag)
+        {
+            return RS_(L"Globals_Language_Default");
+        }
+
+        winrt::Windows::Globalization::Language language(tag);
+        return language.NativeName();
+    }
+
+    // Returns an array of {"und", "en-US", "de-DE", "es-ES", ...}.
+    // "und" is short for "undefined" and is synonym for "Use system language" in this code.
+    winrt::Windows::Foundation::Collections::IObservableVector<winrt::hstring> GlobalAppearance::LanguageList()
+    {
+        // The following code achieves the following:
+        // [1] Get all possible languages we want to allow the user to choose
+        // [2] Sort languages by their ASCII tags, forcing the UI in a consistent/stable order
+        //     I wanted to sort the localized language names initially, but it turned out to be complex.
+        // [3] Remove duplicates between the array of tagSources from 1.
+        //     We don't want to have en-US twice in the list, do we?
+        // [4] Optionally remove unwanted language tags.
+
+        std::vector<winrt::hstring> tags;
+
+        // [1]:
+        {
+            // ManifestLanguages contains languages the app ships with.
+            //
+            // Languages is a computed list that merges the ManifestLanguages with the
+            // user's ranked list of preferred languages taken from the system settings.
+            // As is tradition the API documentation is incomplete though, as it can also
+            // contain regional language variants. If our app supports en-US, but the user
+            // has en-GB or en-DE in their system's preferred language list, Languages will
+            // contain those as well, as they're variants from a supported language. We should
+            // allow a user to select those, as regional formattings can vary significantly.
+            const std::array tagSources{
+                winrt::Windows::Globalization::ApplicationLanguages::ManifestLanguages(),
+                winrt::Windows::Globalization::ApplicationLanguages::Languages()
+            };
+
+            // tags will hold all the flattened results from tagSources.
+            // We resize() the vector to the proper size in order to efficiently GetMany() all items.
+            tags.resize(std::accumulate(
+                tagSources.begin(),
+                tagSources.end(),
+                // tags[0] will be "und" - the "Use system language" item
+                // tags[1..n] will contain tags from tagSources.
+                // --> totalTags is offset by 1
+                1,
+                [](uint32_t sum, const auto& v) -> uint32_t {
+                    return sum + v.Size();
+                }));
+
+            // The first item is always "Use system language"
+            auto data = tags.data();
+            *data++ = systemLanguageTag;

```

Comment:
do you actually enjoy having these weird multi-control settings that enable/disable controls based on the option? I love that this is a simple dropdown option :grin:

Reply:
It's more of a consistency thing. But I guess we already do the same with some settings (i.e. font weight). Like, I guess the rule of thumb should be "if the setting has multiple options (i.e. combo box or radio button representation), add it to the control; otherwise, add a text box". So eh, just ignore this one.